packages/babel-core/src/errors/rewrite-stack-trace.ts

```diff
@@ -0,0 +1,171 @@
+/**
+ * This file uses the iternal V8 Stack Trace API (https://v8.dev/docs/stack-trace-api)
+ * to provide utilities to rewrite the stack trace.
+ * When this API is not present, all the functions in this file become noops.
+ *
+ * beginHiddenCallStack(fn) and endHiddenCallStack(fn) wrap their parameter to
+ * mark an hidden portion of the stack trace. The function passed to
+ * beginHiddenCallStack is the first hidden function, while the function passed
+ * to endHiddenCallStack is the first shown function.
+ *
+ * When an error is thrown _outside_ of the hidden zone, everything between
+ * beginHiddenCallStack and endHiddenCallStack will not be shown.
+ * If an error is thrown _inside_ the hidden zone, then the whole stack trace
+ * will be visible: this is to avoid hiding real bugs.
+ * However, if an error inside the hidden zone is expected, it can be marked
+ * with the expectedError(error) function to keep the hidden frames hidden.
+ *
+ * Consider this call stack (the outer function is the bottom one):
+ *
+ *   1. a()
+ *   2. endHiddenCallStack(b)()
+ *   3. c()
+ *   4. beginHiddenCallStack(d)()
+ *   5. e()
+ *   6. f()
+ *
+ * - If a() throws an error, then its shown call stack will be "a, b, e, f"
+ * - If b() throws an error, then its shown call stack will be "b, e, f"
+ * - If c() throws an expected error, then its shown call stack will be "e, f"
+ * - If c() throws an unexpected error, then its shown call stack will be "c, d, e, f"
+ * - If d() throws an expected error, then its shown call stack will be "e, f"
+ * - If d() throws an unexpected error, then its shown call stack will be "d, e, f"
+ * - If e() throws an error, then its shown call stack will be "e, f"
+ *
+ * Additionally, an error can inject additional "virtual" stack frames using the
+ * injcectVirtualStackFrame(error, filename) function: those are added on the top
+ * of the existig stack, after hiding the possibly hidden frames.
+ * In the example above, if we called injcectVirtualStackFrame(error, "h") on the
+ * expected error thrown by c(), it's shown call stack would have been "h, e, f".
```

Comment:
```suggestion
 * expected error thrown by c(), its shown call stack would have been "h, e, f".
```
May extend the example a bit, for example, what will be the shown call stack if we called `injcectVirtualStackFrame(error, "h")` and `injcectVirtualStackFrame(error, "i")`?